config/config.go

```diff
@@ -217,7 +216,7 @@ type ScrapeConfig struct {
 	// Indicator whether the scraped metrics should remain unmodified.
 	HonorLabels bool `yaml:"honor_labels,omitempty"`
 	// A set of query parameters with which the target is scraped.
-	Params url.Values `yaml:"params,omitempty"`
+	Params map[string]string `yaml:"params,omitempty"`
```

Comment:
That's already way better. I know these are small details but they do have
impact.

So I'd still prefer the consistent concatenating. One could probably find a
way but here complexity might indeed not justify consistency.

On Wed, Jul 29, 2015, 5:42 PM Brian Brazil <notifications@github.com> wrote:

> In config/config.go
> <https://github.com/prometheus/prometheus/pull/934#discussion_r35774634>:
>
> > @@ -217,7 +216,7 @@ type ScrapeConfig struct {
> >  	// Indicator whether the scraped metrics should remain unmodified.
> >  	HonorLabels bool `yaml:"honor_labels,omitempty"`
> >  	// A set of query parameters with which the target is scraped.
> > -	Params url.Values `yaml:"params,omitempty"`
> > +	Params map[string]string `yaml:"params,omitempty"`
>
> Looks like github hid a question. __param_ is the first parameter (not
> the last as julius initially suggested) so that if we ever want to extend
> this functionality the next parameter will be __param2_ which is
> reasonable for an advanced feature that it's not clear we'll ever need.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/prometheus/prometheus/pull/934/files#r35774634>.
>